### ISSUE TITLE
Dry-run problem with Foreign Keys

### ISSUE THREAD

**Issue by Dasc3er** [cakephp/phinx#1303]
If you have two migrations, one with table definitions and one with foreign keys, running Phinx (0.9.2) with the dry-run option will fire the following error:
```
In Table.php line 642:
  Cannot update a table that doesn't exist!
```

Migration example:
```php
$this->table('groups')
    ->addColumn('name', 'string', ['length' => 50])
->create();

$this->table('users')
    ->addColumn('username', 'string')
    ->addColumn('email', 'string', ['length' => 50])
    ->addColumn('password', 'string')
    ->addColumn('group_id', 'integer')
    ->addIndex(['group_id'])
->create();
```

Separate statement (in the same migration or in another one):
```php
$this->table('users')
    ->addForeignKey('group_id', 'groups', 'id')
->update();
```

-----------------

The problem will not appear if you merge the table and foreign key definitions in this way:
```php
$this->table('users')
    ->addColumn('username', 'string')
    ->addColumn('email', 'string', ['length' => 50])
    ->addColumn('password', 'string')
    ->addColumn('group_id', 'integer')
    ->addIndex(['group_id'])

    ->addForeignKey('group_id', 'groups', 'id')
->create();
```

**Comment by dereuromark** [comment]
I ran into a similar issue here.

Also refs https://github.com/cakephp/phinx/issues/1249

**Comment by lorenzo** [comment]
This is fixed in 0.10.0
